core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregationBuilder.java

```diff
@@ -192,29 +192,33 @@ public TermsAggregationBuilder shardMinDocCount(long shardMinDocCount) {
     /**
      * Sets the order in which the buckets will be returned.
      */
-    public TermsAggregationBuilder order(Terms.Order order) {
+    public TermsAggregationBuilder order(BucketOrder order) {
         if (order == null) {
             throw new IllegalArgumentException("[order] must not be null: [" + name + "]");
         }
-        this.order = order;
+        if(order instanceof CompoundOrder || InternalOrder.isKeyOrder(order)) {
+            this.order = order;
+        } else {
+            this.order = BucketOrder.compound(order);
+        }
         return this;
     }
 
     /**
      * Sets the order in which the buckets will be returned.
      */
-    public TermsAggregationBuilder order(List<Terms.Order> orders) {
+    public TermsAggregationBuilder order(List<BucketOrder> orders) {
         if (orders == null) {
             throw new IllegalArgumentException("[orders] must not be null: [" + name + "]");
         }
-        order(Terms.Order.compound(orders));
+        order(orders.size() > 1 ? BucketOrder.compound(orders) : orders.get(0));
```

Comment:
Also I am not against making this method `private` (only called by parser). That way only the one `order(BucketOrder order)` method is exposed to the user to avoid confusion between `List<BucketOrder>` and `BucketOrder.compound(BucketOrder...)`